sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -254,3 +254,52 @@ case class Pow(left: Expression, right: Expression)
       """
   }
 }
+
+object Logarithm {
+  def apply(child: Expression): Expression = new Log(child)
+}
+
+case class Logarithm(left: Expression, right: Expression)
+  extends BinaryMathExpression((c1, c2) => math.log(c2) / math.log(c1), "LOG") {
```

Comment:
I think `BinaryArithmetic` is more limited in its semantic meaning. As I can tell, it is more suitable for the binary expression between two expressions of same type. `BinaryMathExpression` is more like to represent math function with two expressions.

Due to support the case when the left is null in that 10 base logarithm is applied, to override `eval()` is needed here.